src/App/Pages/Accounts/TwoFactorPageViewModel.cs

```diff
@@ -208,7 +208,7 @@ public async Task SubmitAsync()
                 _broadcasterService.Unsubscribe(nameof(TwoFactorPage));
                 var disableFavicon = await _storageService.GetAsync<bool?>(Constants.DisableFaviconKey);
                 await _stateService.SaveAsync(Constants.DisableFaviconKey, disableFavicon.GetValueOrDefault());
-                Device.BeginInvokeOnMainThread(() => { Application.Current.MainPage = new TabsPage(); });
+                Application.Current.MainPage = new TabsPage();
```

Comment:
Let's just add it to the `await SubmitAsync()` call from the `page.DuoWebView.RegisterAction` callback above.